ui/core/src/model/calculations.ts

```diff
@@ -56,38 +60,52 @@ export const CALCULATIONS_CONFIG: Readonly<Record<CalculationType, CalculationCo
 
 export const DEFAULT_CALCULATION: CalculationType = 'Sum';
 
-function first(values: TimeSeriesValueTuple[]): number | undefined {
+type CalculationValue = number | null | undefined;
+
+function first(values: TimeSeriesValueTuple[]): CalculationValue {
   const tuple = values[0];
   return tuple === undefined ? undefined : getValue(tuple);
 }
 
-function last(values: TimeSeriesValueTuple[]): number | undefined {
+function last(values: TimeSeriesValueTuple[]): CalculationValue {
   if (values.length <= 0) return undefined;
 
   const tuple = values[values.length - 1];
   return tuple === undefined ? undefined : getValue(tuple);
 }
 
-function lastNumber(values: TimeSeriesValueTuple[]): number | undefined {
-  const tuple = findLast(values, (tuple) => isNaN(getValue(tuple)) === false);
+function firstNumber(values: TimeSeriesValueTuple[]): CalculationValue {
+  const tuple = values.find((tuple) => typeof getValue(tuple) === 'number');
+  return tuple === undefined ? undefined : getValue(tuple);
+}
+
+function lastNumber(values: TimeSeriesValueTuple[]): CalculationValue {
+  const tuple = findLast(values, (tuple) => typeof getValue(tuple) === 'number');
   return tuple === undefined ? undefined : getValue(tuple);
 }
 
-function mean(values: TimeSeriesValueTuple[]): number | undefined {
+function mean(values: TimeSeriesValueTuple[]): CalculationValue {
   if (values.length <= 0) return undefined;
-  return meanBy(values, getValue);
+  return lodashMean(getNonNullValues(values));
 }
 
-function sum(values: TimeSeriesValueTuple[]): number | undefined {
+function sum(values: TimeSeriesValueTuple[]): CalculationValue {
   if (values.length <= 0) return undefined;
-  return sumBy(values, getValue);
+
```

Comment:
Previous, this would return `NaN` if there were any null values. Now it will sum the non-null values.